runtime/lua/vim/filetype/detect.lua

```diff
@@ -103,7 +107,7 @@ local visual_basic_content = { 'vb_name', 'begin vb%.form', 'begin vb%.mdiform',
 function M.bas(bufnr)
   if vim.g.filetype_bas then
     vim.bo[bufnr].filetype = vim.g.filetype_bas
```

Comment:
When returning the filetype from the function you do not also need to set it (it gets set in `vim.filetype.match()`)
```suggestion
```

Reply:
Ok got it, I wasn't sure if it'd already work like that